ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/Transaction.java

```diff
@@ -571,7 +579,18 @@ private static Bytes32 computeSenderRecoveryHash(
         break;
       case ACCESS_LIST:
         preimage =
-            accessListPreimage(nonce, gasPrice, gasLimit, to, value, payload, accessList, chainId);
+            accessListPreimage(
+                nonce,
+                gasPrice,
+                gasLimit,
+                to,
+                value,
+                payload,
+                accessList.orElseThrow(
```

Comment:
Would this read better as a `checkState` guard before the call to accessListPreimage?